internal/controller/validation/common.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 Apache Software Foundation (ASF)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/apache/incubator-kie-kogito-serverless-operator/internal/controller/platform"
+	"github.com/apache/incubator-kie-kogito-serverless-operator/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var kindRegistryName = func(ctx context.Context) (string, error) {
```

Comment:
Why does it need to be a var?

Reply:
@ricardozanini for mock purpose, to resolve kind-registry in the tests